src/global/global_cuda.h

```diff
@@ -116,6 +116,20 @@ __device__ double atomicAdd(double *address, double val)
 }
     #endif
 
+  // This helper function exists to easily enable/disable printfs inside kernels
+  // And makes it easier to find printfs inside kernels
+
+    #ifndef DISABLE_KERNEL_PRINTF
```

Comment:
Won't this lead to issues in `printf`s that aren't in kernels? Or is the compiler smart enough to overload it?

I'm not sure that this has utility. `printf`s in kernels tend to produce a TON of output and really slow down a simulation so they're pretty hard to accidentally leave enabled and I think we should discourage people from using them.

Reply:
I just saw all the `printf`s in the feedback code. See my comment there about them

Reply:
> Won't this lead to issues in printfs that aren't in kernels? Or is the compiler smart enough to overload it?

No, ```#define kernel_printf printf``` replaces ```kernel_printf``` with ```printf``` as default behavior, and replaces it with a do-nothing function to disable printfs. It does not touch printfs elsewhere.